examples/python/pytket-qujax_qaoa.py

```diff
@@ -0,0 +1,130 @@
+
+print('Gates:\t', hamiltonian_gates)
+print('Qubits:\t', hamiltonian_qubit_inds)
+print('Coefficients:\t', coefficients)
+
+# ## Variational Circuit
+# Before constructing the circuit, we still need to select the mixing Hamiltonian. In our case, we will be using $X$ gates in each qubit, so $H_B = \sum_{i=1}^{n}X_i$, where $n$ is the number of qubits. Notice that the unitary $U(\beta)$, given this mixing Hamiltonian, is an $X$ rotation in each qubit with angle $\beta$.
+# As for the unitary corresponding to the problem Hamiltonian, $U(\gamma)$, it has the following form:
+# $$U(\gamma)=\prod_{(i, j) \in E}e^{-i\gamma\alpha_{ij}Z_iZ_j}$$ 
+# The operation $e^{-i\gamma\alpha_{ij}Z_iZ_j}$ can be performed using two CNOT gates with qubit $i$ as control and qubit $j$ as target and a $Z$ rotation in qubit $j$ in between them, with angle $\gamma\alpha_{ij}$.
+# Finally, the initial state used, in general, with the QAOA is an equal superposition of all the basis states. This can be achieved adding a first layer of Hadamard gates in each qubit at the beginning of the circuit.
+
+# With all the building blocks, let's construct the symbolic circuit using tket. Notice that in order to define the parameters, we use the ```Symbol``` object from the `sympy` package. More info can be found in this [documentation](https://cqcl.github.io/pytket/manual/manual_circuit.html#symbolic-circuits). In order to later convert the circuit to qujax, we need to return the list of symbolic parameters as well.
+
+def qaoa_circuit(n_qubits, depth):
+    circuit = Circuit(n_qubits)
+    p_keys = []
+    
+    # Initial State
+    for i in range(n_qubits):
+        circuit.H(i)
+        
+    for d in range(depth):
+        # Mixing Unitary
+        beta_d = Symbol(f"β_{d}")
+        for i in range(n_qubits):
+            circuit.Rx(beta_d, i)
+        p_keys.append(beta_d)
+        # Hamiltonian
+        gamma_d = Symbol(f"γ_{d}")
+        for index in range(len(hamiltonian_qubit_inds)):
+            pair = hamiltonian_qubit_inds[index]
+            coef = coefficients[index]
+            
+            circuit.CX(pair[0], pair[1])
+            circuit.Rz(gamma_d * coef, pair[1])
+            circuit.CX(pair[0], pair[1])
+            circuit.add_barrier(range(0, n_qubits))
+        p_keys.append(gamma_d)
+        
+    return circuit, p_keys
+
+depth = 3
+circuit, keys = qaoa_circuit(n_qubits, depth)
+
+keys
+
+# Let's check the circuit:
+
+render_circuit_jupyter(circuit)
+
+# # Now for `qujax`
+# The `pytket.extensions.qujax.tk_to_qujax` function will generate a parameters -> statetensor function for us. However, in order to convert a symbolic circuit we first need to define the `symbol_map`. This object maps each symbol key to their corresponding index. In our case, since the object `keys` contains the symbols in the correct order, we can simply construct the dictionary as follows: 
+
+symbol_map = {keys[i]: i for i in range(len(keys))}
```

Comment:
```suggestion
symbol_map = {keys[i]: i for i in keys)}
```

Reply:
I think this is different behaviour

Reply:
Ah sorry - I should have double checked this, my mistake. 

I think you can disregard both the` range(len())` comments